pkg/logs/logs.go

```diff
@@ -30,37 +37,47 @@ func Start() error {
 	return nil
 }
 
-// run sets up the pipeline to process logs and them to Datadog back-end
+// run sets up the pipeline to process logs and send them to Datadog back-end
 func run() {
 	isRunning = true
 
-	cm := sender.NewConnectionManager(
+	connectionManager := sender.NewConnectionManager(
 		config.LogsAgent.GetString("log_dd_url"),
 		config.LogsAgent.GetInt("log_dd_port"),
 		config.LogsAgent.GetBool("dev_mode_no_ssl"),
 	)
 
-	auditorChan := make(chan message.Message, config.ChanSizes)
-	a := auditor.New(auditorChan)
-	a.Start()
-
-	pp := pipeline.NewProvider()
-	pp.Start(cm, auditorChan)
+	messageChan := make(chan message.Message, config.ChanSize)
+	auditor := auditor.New(messageChan)
+	pipelineProvider := pipeline.NewProvider(connectionManager, messageChan)
 
 	sources := config.GetLogsSources()
 
-	l := listener.New(sources.GetValidSources(), pp)
-	l.Start()
-
-	tailingLimit := config.LogsAgent.GetInt("log_open_files_limit")
-	s := tailer.New(sources.GetValidSources(), tailingLimit, pp, a)
-	s.Start()
-
-	c := container.New(sources.GetValidSources(), pp, a)
-	c.Start()
+	networkListeners := listener.New(sources.GetValidSources(), pipelineProvider)
+	containersScanner := container.New(sources.GetValidSources(), pipelineProvider, auditor)
+	filesScanner := tailer.New(
+		sources.GetValidSources(),
+		config.LogsAgent.GetInt("log_open_files_limit"),
+		pipelineProvider,
+		auditor,
+		tailer.DefaultSleepDuration,
+	)
 
+	restart.Start(auditor, pipelineProvider, networkListeners, containersScanner, filesScanner)
```

Comment:
not sure about the name `restart` which seems confusing

Reply:
Agreed, I am still trying to find a better name, any ideas ?